netty/src/main/java/io/grpc/netty/GrpcSslContexts.java

```diff
@@ -219,6 +219,8 @@ public static SslContextBuilder configure(SslContextBuilder builder, Provider jd
         apc = ALPN;
       } else if (JettyTlsUtil.isJettyNpnConfigured()) {
         apc = NPN;
+      } else if (JettyTlsUtil.isJava9AlpnAvailable()) {
```

Comment:
Is it necessary to restrict the use of Java 9's ALPN to when the provider is `SunJSSE`? (I assume that it may always  be SunJSSE now, but couldn't find docs on this)

Reply:
Yes and no. With Jetty, yes, it was necessary. With Java 9, we can do better, by checking whether the Provider supports ALPN (this is done by creating an SSLEngine and trying to use the ALPN API; if it throws, or something like that, it isn't supported).

I want to make the selection logic better for Java 9, but haven't gotten to it. When we're on Java 9+, we can actually us the same selection logic for Conscrypt as SunJSSE, as both support the new API.

But that's for later. For now this hard-coding to SunJSSE is consistent with the approach in `findJdkProvider()`.

Reply:
>For now this hard-coding to SunJSSE is consistent with the approach in findJdkProvider().

Is `findJdkProvider()` or equivalently `GrpcSslContexts.configure(clientContextBuilder, SslProvider.JDK)` not tested in `TlsTest.java`?

Reply:
`findJdkProvider()` is not tested in TlsTest. `configure()` is not tested when using Jetty.

`findJdkProvider()` is hard to test, because it is based on the environment.

Reply:
Agreed.